extensions/modulation.sigmf-ext.md

```diff
@@ -0,0 +1,196 @@
+## Conventions Used in this Document
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+JSON keywords are used as defined in [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).
+
+Augmented Backus-Naur form (ABNF) is used as defined by [RFC
+5234](https://tools.ietf.org/html/rfc5234) and updated by [RFC
+7405](https://tools.ietf.org/html/rfc7405).
+
+Fields defined as "human-readable", a "string", or simply as "text" shall be
+treated as plaintext where whitespace is significant, unless otherwise
+specified.
+
+## Specification
+
+This specification defines an extension `modulation` namespace
+that can be used in SigMF `annotations` to describe wireless communications
+systems.
+
+### Annotations
+
+This extension adds the following field to the `annotations` global SigMF object:
+
+|name|required|type|description|
+|----|--------------|-------|-----------|
+|modulation|false|object|Describes a communications system.|
+
+The field of communications is vast, and there may be communications systems that cannot be described using the names and fields described in this extension. If you need additional or different fields to describe a system, create a new extension that adds the necessary fields to the `modulation` namespace and/or submit the new fields to be upstreamed into this specification.
+
+#### The `modulation` Object
+
+`modulation` objects contain name/value pairs that describe the modulations of communications systems. These objects MAY contain the following pairs:
+
+|name|required|type|
+|----|--------------|-------|
+|`type`|false|string|
+|`class`|false|string|
+|`carrier_variant`|false|string|
+|`symbol_variant`|false|string|
+|`order`|false|uint|
+|`duplexing`|false|string|
+|`multiplexing`|false|string|
+|`multiple_access`|false|string|
+|`spreading`|false|string|
+|`bandwidth`|false|float|
+|`system`|false|string|
+
+###### The `type` Pair
+
+The `type` name can have the following values:
+
+|value|description|
+|----|-------|
+|`analog`|analog modulation scheme|
+|`digital`|digital modulation scheme|
+
+###### The `class` Pair
+
+The `class` name can have the following values:
+
+|value|description|
+|----|-------|
+|`am`|(analog) amplitude modulation|
+|`fm`|(analog) frequency modulation|
+|`pm`|(analog) phase modulation|
+|`ssb`|(analog) single side-band|
+|`dsb`|(analog) dual side-band|
+|`vsb`|(analog) vestigal side-band|
+|`ask`|(analog) amplitude-shift keying|
+|`fsk`|(digital) frequency-shift keying|
+|`psk`|(digital) phase-shift keying|
+|`qam`|(digital) quadrature-amplitude modulation|
+|`ook`|(digital) on-off keying|
+|`cpm`|(digital) continuous phase modulation|
+|`msk`|(digital) minimum-shift keying|
+
+###### The `carrier_variant` pair
+
+The `carrier_variant` name can have the following values:
```

Comment:
Unless someone with the need for this to be described separately this is either overly complex and incomplete or part of the modulation. For example, is there anything other than AM that would use a suppressed carrier? In that case we should just have a modulation type that is AM-SC. In the multi-carrier (OFDM) case the `class` should be OFDM because none of the other classes make sense.